node-js-basics-101/node_modules/asynquence-contrib/contrib.src.js

```diff
@@ -0,0 +1,2360 @@
+						handlers.push(ASQ.iterable().then(function $$then() {
+							resolved = true;
+
+							// mark all handlers across this `alts(..)` as resolved now
+							handlers = handlers.filter(function $$filter(handler) {
+								return !(handler.resolved = true);
+							});
+
+							// channel still open?
+							if (!channel.closed) {
+								isq.next({ value: channel.messages.shift(), channel: channel });
+							}
+							// channel already closed?
+							else {
+									isq.next({ value: ASQ.csp.CLOSED, channel: channel });
+								}
+						}));
+
+						// queue up take handler
+						channel.take_queue.push(handlers[handlers.length - 1]);
+
+						// put waiting on this queued take?
+						if (channel.put_queue.length > 0) {
+							schedule(function handleUnblocking() {
+								if (!resolved) {
+									unblock(channel.put_queue.shift());
+									unblock(channel.take_queue.shift());
+								}
+							});
+						}
+					}
+			})(open[i]);
+		}
+
+		return ret;
+	}
+
+	function altsAsync(chans, cb) {
+		var ret = ASQ(alts(chans));
+
+		if (cb && typeof cb == "function") {
+			ret.pThen(cb, cb);
+		} else {
+			return ret;
+		}
+	}
+
+	function timeout(delay) {
+		var ch = channel();
+		setTimeout(ch.close, delay);
+		return ch;
+	}
+
+	function go(gen, args) {
+		// goroutine arguments passed?
+		if (arguments.length > 1) {
+			if (!args || !Array.isArray(args)) {
+				args = [args];
+			}
+		} else {
+			args = [];
+		}
+
+		return regeneratorRuntime.mark(function $$go(token) {
+			var unblock, ret, msg, err, type, done, it;
+			return regeneratorRuntime.wrap(function $$go$(context$3$0) {
+				while (1) switch (context$3$0.prev = context$3$0.next) {
+					case 0:
+						unblock = function unblock() {
+							token.unblock_count++;
+
+							if (token.block && !token.block.marked) {
+								token.block.marked = true;
+								token.block.next();
+							}
+						};
+
+						done = false;
+
+						// keep track of how many requests for unblocking
+						// have occurred
+						token.unblock_count = token.unblock_count || 0;
+
+						// keep track of how many goroutines are running
+						// so we can infer when we're done go'ing
+						token.go_count = (token.go_count || 0) + 1;
+
+						// need to initialize a set of goroutines?
+						if (token.go_count === 1) {
+							// create a default channel for these goroutines
+							token.channel = channel();
+							token.channel.messages = token.messages;
+							token.channel.go = function $$go() {
+								// add the goroutine (called with any args) to
+								// the handling queue
+								token.add(go.apply(ø, arguments));
+								// unblock the goroutine handling for this
+								// new goroutine
+								unblock();
+							};
+							// starting out with initial channel messages?
+							if (token.channel.messages.length > 0) {
+								// fake back-pressure blocking for each
+								token.channel.put_queue = token.channel.messages.map(function $$map() {
```

Comment:
__[jshint]__
* Functions declared within loops referencing an outer scoped variable may lead to confusing semantics.

<!-- Comment by [SideCI](https://sideci.com) -->